packages/ai-jsx/src/core/image-gen.tsx

```diff
@@ -77,14 +79,46 @@ export function ImageGenProvider<T extends ImageGenPropsWithChildren>(
  *    </ImageGen>
  * ```
  */
-export function ImageGen(
-  { children, ...props }: ImageGenPropsWithChildren & Record<string, unknown>,
-  { getContext }: RenderContext
-) {
+export function ImageGen({ children, ...props }: ImageGenPropsWithChildren, { getContext }: RenderContext) {
   const [ImageGenComponent, defaultProps] = getContext(imageGenContext);
   return (
     <ImageGenComponent {...defaultProps} {...props}>
       {children}
     </ImageGenComponent>
   );
 }
+
+/**
+ * A wrapper around {@link ImageGen} that renders the generated image(s) as an HTML "img" tag.
+ * To be used only in the browser, not the CLI.
+ *
+ * @returns HTML "img" tag(s) with the generated image(s).
+ */
+export async function ImageGenHTML(
+  {
+    children,
+    width = undefined,
+    height = undefined,
+    alt = undefined,
+    ...props
+  }: ImageGenPropsWithChildren & {
+    /** Width for the HTML image tag. */ width?: string;
+    /** Height for the HTML image tag. */ height?: string;
+    /** Alternative text for the HTML image tag.  */ alt?: string;
+  },
+  { render }: RenderContext
+) {
+  const urls = await render(<ImageGen {...props}>{children}</ImageGen>);
+  return (
+    <AIR.React>
+      {urls.split('\n').map((url) =>
+        ReactModule.createElement('img', {
+          src: url,
+          alt: alt ?? 'Generated image',
```

Comment:
If we happened to know the prompt, it would be nice to put here. But that's probably too tricky to wire up.

Reply:
If we had an abstract image representation I think this would be straightforward :)

Reply:
Done. The `<Image>` tag now holds a `prompt` and a `modelName` property. These are never rendered however and can only be used for debugging at the moment.